client/src/js/services/grid/GridGrouping.js

```diff
@@ -98,9 +99,32 @@ function GridGroupingService(GridAggregators, uiGridGroupingConstants, Session,
     }
   }
 
-  function handleBatchSelection (){
+  function handleBatchSelection() {
     var gridApi = this.gridApi;
+    var gridRows = gridApi.selection.getSelectedGridRows();
+    var parents = {};
+
+    var hasSelections = gridApi.selection.getSelectedRows().length > 0;
+
+    gridRows.forEach(function (row) {
+      var parentRow = row.treeNode.parentRow;
+      if (parentRow.treeLevel === 0 && !parents[parentRow.uid]) {
+        parentRow.isSelected = true;
+        parents[parentRow.uid] = parentRow;
+        SELECTED_GROUP_HEADERS = parents;
```

Comment:
Aha, that makes sense.  I don't think there is a clearer way of expressing what you wrote.  LGTM :+1: